q2_diversity_lib/alpha.py

```diff
@@ -6,31 +6,22 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
-import biom
 import pandas as pd
 import skbio.diversity
+import biom
+from unifrac import faith_pd as f_pd
 
-from ._util import _drop_undefined_samples, _disallow_empty_tables
+from q2_types.feature_table import BIOMV210Format
+from q2_types.tree import NewickFormat
+from ._util import (_drop_undefined_samples,
+                    _disallow_empty_tables)
 
 
 @_disallow_empty_tables
-def faith_pd(table: biom.Table, phylogeny: skbio.TreeNode) -> pd.Series:
-    presence_absence_table = table.pa()
-    counts = presence_absence_table.matrix_data.toarray().astype(int).T
-    sample_ids = presence_absence_table.ids(axis='sample')
-    feature_ids = presence_absence_table.ids(axis='observation')
-
-    try:
-        result = skbio.diversity.alpha_diversity(metric='faith_pd',
-                                                 counts=counts,
-                                                 ids=sample_ids,
-                                                 otu_ids=feature_ids,
-                                                 tree=phylogeny)
-    except skbio.tree.MissingNodeError as e:
-        message = str(e).replace('otu_ids', 'feature_ids')
-        message = message.replace('tree', 'phylogeny')
-        raise skbio.tree.MissingNodeError(message)
-
+def faith_pd(table: BIOMV210Format, phylogeny: NewickFormat) -> pd.Series:
+    table_str = str(table)
+    tree_str = str(phylogeny)
```

Comment:
Stylistically, this doesn't match what you have done in the unifrac methods, below. I like what you've done in those unifrac methods, by in-lining the `str`, I suggest you do that here,  too.